### ISSUE TITLE
allow data with functions

### ISSUE THREAD

**Issue by dzmitry-lahoda** [fluencelabs/aqua#351]
now:
```
func foobar(progress: string -> ()):
```

allow for
```
data CallbackHacks:
    progress: u8 -> ()
    log: string -> ()
    log_2: string, string -> ()
```

because i cannot express algorithms without callbacks, and at same time i need to propagate callbacks layers deep which is very  hard. so this is ultimate hack `callback: string, []string -> ?[]?string` where first  is function name, and than parameters, and then may be return types

**Comment by alari** [comment]
It's gonna be solved with #33, which comes after #183.

**Comment by dzmitry-lahoda** [comment]
```
log_1_nullable_2_non_nullable
log_1_nullable_2_nullable
```
